pkg/controller/account/account_controller.go

```diff
@@ -188,21 +188,28 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 				awsv1alpha1.AccountCreationFailed,
 				initErr.Error(),
 				"Failed to initialize new CCS account",
-				AccountFailed,
+				v1alpha1.AccountStatusFailed,
 			)
 			if stateErr != nil {
 				reqLogger.Error(stateErr, "failed setting account state", "desiredState", "Failed")
 			}
 			reqLogger.Error(initErr, "failed initializing new CCS account")
 			return result, initErr
 		}
-		utils.SetAccountStatus(currentAcctInstance, AccountCreating, awsv1alpha1.AccountCreating, AccountCreating)
-		updateErr := r.statusUpdate(currentAcctInstance)
-		if updateErr != nil {
+
+		err = utils.SetAccountStatus(
```

Comment:
Maybe instead of updating this util we should consider making this a bit more encapsulated in the account itself?  So instead of the controller telling the account "Hey, you're failed here's your status" it would say "Hey, account, mark yourself as failed for this reason"

Then we can call it something like this `currentAcctInstance.MarkFailed("Account Creation Failed or whatever we want because this is just literally a string to string field but the actual enum logic is handled in the account def itself")`, save the account and exit.

Reply:
This would also give us the benefit of slowly rolling out that change to different pieces, instead of having to do large sweeping refactors, which I think is what bit us last time.

Reply:
@iamkirkbater I've thought about this and think it might be best to place the `MarkFailed` style changes within their own PR once this one has been merged - I don't want to wrap even more content into this PR to make reviewing more complicated

Reply:
That's fine with me.